File: 01_GOF23/DP_examples/CSharp/03_Behavioral/Strategy/Context.cs
namespace Behavioral.Strategy;
public class Context
{
    private IStrategy strategy;
    private static Context instance;
    public static Context GetInstance(IStrategy strategy)
    {
        if (instance == null)
            instance = new Context();
        instance.strategy = strategy;
        return instance;
    }
    public void Sort(int[] array)
    {
        strategy.Sort(array);
    }
}

File: 01_GOF23/DP_examples/Go/03_Behavioral/Strategy/context.go
package strategy

// Context 上下文类
type Context struct {
	strategy IStrategy
}

// NewContext 创建上下文实例
func NewContext(strategy IStrategy) *Context {
	return &Context{
		strategy: strategy,
	}
}

// SetStrategy 设置策略
func (c *Context) SetStrategy(strategy IStrategy) {
	c.strategy = strategy
}

// Sort 执行排序操作
func (c *Context) Sort(array []int) {
	c.strategy.Sort(array)
}
